### ISSUE TITLE
add the ability to throw exceptions on authentication failure

### ISSUE THREAD

**Issue by bdwain** [gonzalo-bulnes/simple_token_authentication#109]
This ability seemed to be referenced in a pull request #61 , but the feature never committed. I found it confusing that saying acts_as_token_authentication_handler didn't cause any errors on a failed authentication in one of my controllers. I wouldn't have any way of knowing that they didn't authenticate unless I tried to reference current_user.

I think a good solution that doesn't break any previous code would be to add a something like this (the first method was taken directly from @donbobka in #61 )

````ruby
    def require_authentication_of_entity!
      self.method(:"#{@@entity.name.singularize.underscore}_signed_in?").call ||
        throw(:warden, scope: :"#{@@entity.name.singularize.underscore}")
    end

    def acts_as_token_authentication_handler_for!(entity, options = {})
       acts_as_token_authentication_handler_for Entity, options
       append_before_filter :require_authentication_of_entity!, options.slice(:only, :except)
    end
````

this would follow the common pattern of having the ! methods throwing exceptions or doing other noteworthy things, just like rails' save/save! methods. I'm going to make a pull request for this, though I'm new to cucumber so any suggestions on the tests would be appreciated. Seems like there shouldn't have to be too much new testing.

**Comment by dhirajbajaj** [comment]
is before_filter implemented. Looks like a important feature to have.

**Comment by gonzalo-bulnes** [comment]
Hi @dhirajbajaj, the feature is not implemented, which doesn't mean you can't use it, see #154!

**Comment by gonzalo-bulnes** [comment]
Hello @bdwain, @dhirajbajaj,

Oops, I forgot to announce here that I released **v1.10.0** with support for the option: `fallback: :exception`, a few weeks ago, please see #174 for details : )

Cheers!

**Comment by dhirajbajaj** [comment]
:+1:
